crates/extensions/tedge_log_manager/src/actor.rs

```diff
@@ -0,0 +1,269 @@
+                self.config.current_operations.remove(&topic.name);
+                error!("{}", error_message);
+                Ok(())
+            }
+        }
+    }
+
+    /// executes the log file request
+    ///
+    /// - sends request executing (mqtt)
+    /// - uploads log content (http)
+    /// - sends request successful (mqtt)
+    async fn execute_logfile_request_operation(
+        &mut self,
+        topic: &Topic,
+        request: &LogRequestPayload,
+    ) -> Result<(), LogManagementError> {
+        if !request.status.eq(&CommandStatus::Executing) {
+            self.mqtt_publisher
+                .send(create_operation_message(
+                    topic,
+                    request,
+                    CommandStatus::Executing,
+                ))
+                .await?;
+        }
+
+        let log_content = log_manager::new_read_logs(
+            &self.plugin_config.files,
+            &request.log.log_type,
+            request.log.date_from,
+            request.log.lines,
+            &request.log.search_text,
+        )?;
+
+        self.send_log_file_http(log_content, request.tedge_url.clone())
+            .await?;
+
+        self.mqtt_publisher
+            .send(create_operation_message(
+                topic,
+                request,
+                CommandStatus::Successful,
+            ))
+            .await?;
+
+        self.config.current_operations.remove(&topic.name);
+        info!("Log request processed.");
+        Ok(())
+    }
+
+    pub async fn send_log_file_http(
+        &mut self,
+        log_content: String,
+        url: String,
+    ) -> Result<(), LogManagementError> {
+        let req_builder = HttpRequestBuilder::put(url)
+            .header("Content-Type", "text/plain")
+            .body(log_content);
+
+        let request = req_builder.build()?;
+
+        let http_result = match self.http_proxy.await_response(request).await? {
+            Ok(response) => match response.status() {
+                StatusCode::OK | StatusCode::CREATED => Ok(response),
+                code => Err(HttpError::HttpStatusError(code)),
+            },
+            Err(err) => Err(err),
+        };
+
+        let _ = http_result.error_for_status()?;
+        info!("Logfile uploaded.");
+        Ok(())
+    }
+
+    pub async fn process_file_watch_events(
+        &mut self,
+        event: FsWatchEvent,
+    ) -> Result<(), LogManagementError> {
+        let path = match event {
+            FsWatchEvent::Modified(path) => path,
+            FsWatchEvent::FileDeleted(path) => path,
+            FsWatchEvent::FileCreated(path) => path,
+            FsWatchEvent::DirectoryDeleted(_) => return Ok(()),
+            FsWatchEvent::DirectoryCreated(_) => return Ok(()),
+        };
+
+        if path
+            .file_name()
+            .ok_or_else(|| PathsError::ParentDirNotFound {
+                path: path.as_os_str().into(),
+            })?
+            .eq("tedge-log-plugin.toml")
```

Comment:
Use the variable instead of hardcode.

```suggestion
            .eq(DEFAULT_PLUGIN_CONFIG_FILE_NAME)
```